discovery-provider/src/queries/get_user_listening_history.py

```diff
@@ -57,34 +67,59 @@ def _get_user_listening_history(session: Session, args: GetUserListeningHistoryA
     if not listening_history_results:
         return []
 
-    # add query pagination
-    listening_history_results = listening_history_results[offset : offset + limit]
-
+    # Map out all track ids and listen dates
     track_ids = []
-    listen_dates = []
+    listen_dates = {}
     for listen in listening_history_results:
         track_ids.append(listen["track_id"])
-        listen_dates.append(listen["timestamp"])
-
-    track_results = (session.query(Track).filter(Track.track_id.in_(track_ids))).all()
```

Comment:
there was no "is_current" filter here - I image this causing the query to be  slow